File: scripts/agent-bench/fixture-ts/src/dedupe.ts
import type { DocEntry } from "./types.js";

/** Keep the best-scoring hit per path (input must already be sorted best-first). */
export function collapsePaths(hits: DocEntry[]): DocEntry[] {
  const seen = new Set<string>();
  return hits.filter((h) => {
    if (seen.has(h.path)) return false;
    seen.add(h.path);
    return true;
  });
}

File: scripts/agent-bench/fixture-rust/src/dedupe.rs
use crate::types::DocEntry;
use std::collections::HashSet;

/// Keep the best-scoring hit per path (input must already be sorted best-first).
pub fn collapse_paths(hits: Vec<DocEntry>) -> Vec<DocEntry> {
    let mut seen: HashSet<String> = HashSet::new();
    hits.into_iter().filter(|h| seen.insert(h.path.clone())).collect()
}
